codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationRegistryGenerator.kt

```diff
@@ -108,146 +111,146 @@ class ServerOperationRegistryGenerator(
+                    }
                 }
                 """,
                 *codegenScope
             )
         }
     }
 
-    /*
-     * Renders the `OperationRegistryBuilder` implementation, where operations are stored.
+    /**
+     * Renders the `OperationRegistryBuilder`'s impl block, where operations are stored.
      * The `build()` method converts the builder into an `OperationRegistry` instance.
      */
     private fun renderOperationRegistryBuilderImplementation(writer: RustWriter) {
-        // A lot of things can become pretty complex in this type as it will hold 2 generics per operation.
-        Attribute.Custom("allow(clippy::all)").render(writer)
-        writer.rustBlockTemplate(
-            """
-            impl<$genericArguments> $operationRegistryBuilderNameWithArguments
-            """.trimIndent(),
-            *codegenScope
-        ) {
-            val registerOperations = operationNames.mapIndexed { i, operationName ->
-                """pub fn $operationName(self, value: Op$i) -> Self {
-                let mut new = self;
-                new.$operationName = Some(value);
-                new
-                }"""
-            }.joinToString(separator = "\n")
-            val registerOperationsBuilder = operationNames.map { operationName ->
-                """
-                $operationName: match self.$operationName {
-                    Some(v) => v,
-                    None => return Err(${operationRegistryBuilderName}Error::UninitializedField(${operationName.dq()})),
-                }
-                """
-            }.joinToString(separator = "\n,")
-            rustTemplate(
-                """
-                $registerOperations
-                pub fn build(self) -> Result<$operationRegistryNameWithArguments, ${operationRegistryBuilderName}Error> {
-                    Ok($operationRegistryName { $registerOperationsBuilder, _phantom: #{Phantom} })
+        writer.rustBlock("impl<$genericArguments> $operationRegistryBuilderNameWithArguments") {
+            operationNames.forEachIndexed { i, operationName ->
+                rust(
+                    """
+                    pub fn $operationName(self, value: Op$i) -> Self {
+                        let mut new = self;
+                        new.$operationName = Some(value);
+                        new
+                    }
+                    """
+                )
+            }
+
+            rustBlock("pub fn build(self) -> Result<$operationRegistryNameWithArguments, ${operationRegistryErrorName}>") {
+                withBlock("Ok( $operationRegistryName {", "})") {
+                    for (operationName in operationNames) {
+                        rust(
+                            """
+                            $operationName: match self.$operationName {
+                                Some(v) => v,
+                                None => return Err(${operationRegistryErrorName}::UninitializedField("operationName")),
```

Comment:
There is an error here:

```suggestion
                                None => return Err(${operationRegistryErrorName}::UninitializedField("$operationName")),
```

Reply:
Good catch!